build/helper/metadata_add_all.py

```diff
@@ -77,13 +73,27 @@ def _add_is_error_handling(f):
 
 def _add_buffer_info(parameter):
     '''Adds buffer information to the parameter metadata iff 'size' is defined else assume not a buffer'''
+
+    # We are going to treat ViChar[], ViString, ViConstString and ViRsrc the same. For simplicity in the generated
+    # code, we will treat them as 'ViChar' and is_buffer true
+    t = parameter['type']
+    if (t.find('[ ]') > 0) or (t.find('[]') > 0) or t == 'ViString' or t == 'ViConstString' or t == 'ViRsrc':
+        parameter['type'] = 'ViChar'
+        parameter['original_type'] = t
+        parameter['is_buffer'] = True
+
     try:
         parameter['size']
```

Comment:
FWIW, more pythonic is to say
if 'size' not in parameter:
    # Not populated, assume {'mechanism': 'fixed', 'value': 1}
    ....

Of course this was already that way so you can choose to not fix.

Reply:
I started using if 'something' not in something_else but went with this to be consistent to other things. I figured if I changed all of them to the if way, you'd tell me it goes in a different PR. :)

Reply:
> I figured if I changed all of them to the if way, you'd tell me it goes in a different PR. :)

You're probably right. Easy fix for own PR if someone feels like it.